### PR TITLE
Chiara/websockets to ops

### PR DESCRIPTION
**Pull request by ch-iara**

### CODE REVIEW

File: rita_client/src/heartbeat/mod.rs
Comment (jkilpatr) [inline rita_client/src/heartbeat/mod.rs:131]: just flagging that this shouldn't be checked in in the final pr.

File: rita_client/src/operator_update/ops_websocket.rs
Comment (jkilpatr) [inline rita_client/src/operator_update/ops_websocket.rs:42]: this loop should be split out into its own function so it's easier to separate the logic form all the threads/runners

File: rita_client/src/operator_update/ops_websocket.rs
Comment (jkilpatr) [inline rita_client/src/operator_update/ops_websocket.rs:52]: why exactly do we want to break in this case?

File: rita_client/src/operator_update/ops_websocket.rs
Comment (jkilpatr) [inline rita_client/src/operator_update/ops_websocket.rs:60]: this timeout should be a constant somewhere and should be much lower like 100ms, we're not really waiting for a response from upstream we're checking if one has already arrived.

**Review by jkilpatr — CHANGES_REQUESTED** [review]
This is a great structure overall and only needs some minor cleanup before it's ready to merge.

File: rita_client/src/operator_update/mod.rs
Comment (jkilpatr) [inline rita_client/src/operator_update/mod.rs:106]: these accessory functions would do better in the settings module, also atomic setters would be nice.

Comment (ch-iara) [reply rita_client/src/operator_update/mod.rs:106]: I've put these ones that are just from settings in the settings module, but some of these getters are not easily transferable (see get_hardware_info_update) or aren't really settings (the usage/throughput getters and rita uptime) so I've left them here

File: althea_types/src/websockets.rs
Comment (jkilpatr) [inline althea_types/src/websockets.rs:109]: shouldn't we use the latest update type not the legacy here?

**Review by jkilpatr — APPROVED** [review]
I really like the comments and architecture here, very clean and everything I hoped for when speccing this out. 

Just one comment on what type we use. But we can possibly skip that.

File: althea_types/src/websockets.rs
Comment (ch-iara) [reply althea_types/src/websockets.rs:109]: we do, with UpdateV2- this is left in to maintain ops compatibility since I've just moved this struct over from the other file

**Comment by jkilpatr** [comment]
This is really great work, excited to merge this into the beta22 branch for a firmware build.

**Pull request merged**
